starcluster/awsutils.py

```diff
@@ -597,6 +597,32 @@ def wait_for_propagation(self, instances=None, spot_requests=None,
                 instance_ids, self.get_all_instances, 'instance-id',
                 'instances', max_retries=max_retries, interval=interval)
 
+    def _check_for_propagation(self, obj_ids, fetch_func, id_filter, obj_name):
```

Comment:
Perhaps there is a better name for this? I know I am a little confused by two methods with more or less the same name.